pkg/crypto/signature.go

```diff
@@ -0,0 +1,127 @@
+package crypto
+
+import (
+	"crypto/sha256"
+	"crypto/sha512"
+	"encoding/binary"
+	"encoding/hex"
+	"errors"
+	"fmt"
+
+	"golang.org/x/crypto/ripemd160"
+
+	"github.com/btcsuite/btcd/btcec/v2"
+	"github.com/btcsuite/btcd/btcec/v2/ecdsa"
+	"github.com/icon-project/stacks-go-sdk/pkg/stacks"
+)
+
+type MessageSignature struct {
+	Type StacksMessageType
+	Data string
+}
+
+type StacksMessageType int
+
+const (
+	Address StacksMessageType = iota
+	MessageSignatureType
+)
+
+type StacksPrivateKey struct {
+	Data       []byte
+	Compressed bool
+}
+
+func SignWithKey(privateKey []byte, messageHash string) (MessageSignature, error) {
+	privKey, _ := btcec.PrivKeyFromBytes(privateKey[:32])
+	messageHashBytes, err := hex.DecodeString(messageHash)
+	if err != nil {
+		return MessageSignature{}, err
+	}
+	signature := ecdsa.SignCompact(privKey, messageHashBytes, true)
+
+	recoveryID := signature[0] - 27 - 4
```

Comment:
This could be clarified what 27 and 4 means here. otherwise might as well just write -31

Reply:
Added to PR 17 to avoid rebase and merge conflicts https://github.com/icon-project/stacks-go-sdk/pull/17/commits/5d6bd1fd09579873e8a506b96b77927cce9e85bd